lib/celluloid/eventsource.rb

```diff
@@ -155,12 +155,11 @@ def establish_connection
           end
 
           if parser.status_code != 200
-            until @socket.eof?
-              parser << readline_with_timeout(@socket)
+            # Consume response body if present
+            if !parser.headers["Content-Length"].nil?
+              parser << @socket.read(parser.headers["Content-Length"].to_i)
             end
-            # If the server returns a non-200, we don't want to close-- we just want to
-            # report an error
-            # close
+            close
```

Comment:
*Not* closing the connection was a change that we made a while back. I'm not sure what the benefit of keeping it open was supposed to be. @jkodumal, do you remember?